### ISSUE TITLE
Android S explicitly exported manifest components

### ISSUE THREAD

**Issue by laalto** [microsoft/appcenter-sdk-android#1506]
### **Description**

Android S requires that manifest components with `intent-filter`s explicitly declare `android:exported` value for apps targeting S or above.

https://developer.android.com/about/versions/12/behavior-changes-12#exported

### **Repro Steps**

1. Project with appcenter-distribute 4.1.0
2. Change compileSdkVersion to 'android-S' and targetSdkVerrsion to 'S'
3. Build the project
4. Attempt to install to a device/emulator with Android S preview

Installation fails along the lines of

```
Execution failed for task ':app:installDebug'.
> java.util.concurrent.ExecutionException: com.android.builder.testing.api.DeviceException: com.android.ddmlib.InstallException: INSTALL_PARSE_FAILED_MANIFEST_MALFORMED: Failed parse during installPackageLI: /data/app/vmdl939853150.tmp/base.apk (at Binary XML file line #625): com.microsoft.appcenter.distribute.DeepLinkActivity: Targeting S+ (version 10000 and above) requires that an explicit value for android:exported be defined when intent filters are present
```

Looking up https://github.com/microsoft/appcenter-sdk-android/blob/develop/sdk/appcenter-distribute/src/main/AndroidManifest.xml there seems to also be a `receiver` that would need an explicit `exported` attr value.

**Comment by AnastasiaKubova** [comment]
Hi! Thanks for getting in touch with us! 
I marked it as a bug.

**Comment by igor-khrupin-life360** [comment]
Do we have any updates regarding this bug?

**Comment by laalto** [comment]
As a temporary workaround, I'm locally redeclaring the appcenter components in app-level AndroidManifest.xml with explicit `android:exported` settings to enable Android S testing:

```
<application>

        <activity
            android:exported="true"
            android:name="com.microsoft.appcenter.distribute.DeepLinkActivity"
            android:theme="@android:style/Theme.NoDisplay">
            <intent-filter>
                <action android:name="android.intent.action.VIEW" />

                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />

                <data
                    android:host="updates"
                    android:path="/"
                    android:scheme="appcenter" />
            </intent-filter>
        </activity>

        <receiver
            android:exported="true"
            android:name="com.microsoft.appcenter.distribute.DownloadManagerReceiver">
            <intent-filter>
                <action android:name="android.intent.action.DOWNLOAD_COMPLETE" />
                <action android:name="android.intent.action.DOWNLOAD_NOTIFICATION_CLICKED" />
            </intent-filter>
        </receiver>

    </application>

```

**Comment by igor-khrupin-life360** [comment]
@laalto I did the same thing and using my fork as a dependency. I'm curious when we will be able to use at least a SNAPSHOT version of the lib. 
Also, I have created PR with that fix. You can find it above.

**Comment by AnastasiaKubova** [comment]
We have already created a ticket about this issue. We will notify you when we will have any updates.

**Comment by AnastasiaKubova** [comment]
The fix has been released in the [4.2.0](https://github.com/microsoft/appcenter-sdk-android/releases/tag/4.2.0) version of App Center SDK. Please update the App Center SDK to the latest version and let me know if this issue still persists.
